src/overlays/actors/ovl_Bg_Mori_Idomizu/z_bg_mori_idomizu.c

```diff
@@ -20,23 +35,152 @@ const ActorInit Bg_Mori_Idomizu_InitVars = {
+        this->targetWaterLevel = -282.0f;
+    } else {
+        this->targetWaterLevel = 184.0f;
+    }
+    if (switchFlag && !this->prevSwitchFlag) {
+        func_800800F8(globalCtx, 0xCA8, 0x46, thisx, 0);
+        this->drainTimer = 90;
+    } else if (!switchFlag && this->prevSwitchFlag) {
+        func_800800F8(globalCtx, 0xCA8, 0x46, thisx, 0);
+        this->drainTimer = 90;
+        thisx->posRot.pos.y = 0.0f;
+    }
+    this->drainTimer--;
+    if ((roomNum == 7) || (roomNum == 8) || (roomNum == 9)) {
+        if (this->drainTimer < 70) {
+            Math_ApproxF(&thisx->posRot.pos.y, this->targetWaterLevel, 3.5f);
+            BgMoriIdomizu_SetWaterLevel(globalCtx, thisx->posRot.pos.y);
+            if (this->drainTimer > 0) {
+                if (switchFlag) {
+                    func_800788CC(NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+                } else {
+                    func_800788CC(NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+                }
+            }
+        }
+    } else {
+        thisx->posRot.pos.y = this->targetWaterLevel;
+        BgMoriIdomizu_SetWaterLevel(globalCtx, thisx->posRot.pos.y);
+        Actor_Kill(thisx);
+        return;
+    }
+    this->prevSwitchFlag = switchFlag;
+}
+
+void BgMoriIdomizu_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgMoriIdomizu* this = THIS;
+
+    if (this->actionFunc != NULL) {
+        this->actionFunc(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/BgMoriIdomizu_Init.s")
+void BgMoriIdomizu_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgMoriIdomizu* this = THIS;
+    u32 gFrames = globalCtx->gameplayFrames;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/BgMoriIdomizu_Destroy.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_mori_idomizu.c", 356);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A47D8.s")
+    func_80093D84(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A47FC.s")
+    gSPMatrix(oGfxCtx->polyXlu.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_mori_idomizu.c", 360),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A4850.s")
+    gSPSegment(oGfxCtx->polyXlu.p++, 0x08, globalCtx->objectCtx.status[this->moriTexObjIndex].segment);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A4874.s")
+    gDPSetEnvColor(oGfxCtx->polyXlu.p++, 0, 0, 0, 128);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/BgMoriIdomizu_Update.s")
+    gSPSegment(oGfxCtx->polyXlu.p++, 0x09,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0x7F - (gFrames & 0x7F), gFrames % 0x80, 0x20, 0x20, 1,
+                                gFrames & 0x7F, gFrames % 0x80, 0x20, 0x20));
```

Comment:
It doesn't really matter whether we use `& 0x7F` or `% 0x80` but imo it should be consistent in the same function at least.

Reply:
You'd think that, but it doesn't actually match if you do.